### PR TITLE
Changed Method of Error Handling

### PR DESCRIPTION
**Pull request by rakaar**
- Used native node js `Error` object to handle errors
- In case empty rows returning the message `{ 'errror': 'no data' }`

### CODE REVIEW

**Comment by rakaar** [comment]
Fixes #5

File: index.js [hunk 1]
```diff
@@ -24,12 +24,14 @@ app.get('/:sheet', (req, res) => {
   let doc = new GoogleSpreadsheet(sheet_id);
   let sheet_data = [];
   doc.useServiceAccountAuth(creds, err => {
-    if (err) res.send(err);
-    doc.getRows(1, (err, rows) => {
-      if (rows[0] == undefined) {
-        return res.send('');
-      }
-      if (err) res.send(err);
+    if (err) throw new Error('Unable to connect to docs');
```
Comment (thealphadollar) [inline index.js]: These errors should be returned to the frontend IMO with the exact message as you wrote for throwing error. Setting the status code as 500 would make the catch block active on the frontend and we will have the error console logged as well as have a response in the frontend.

**Review by thealphadollar — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (rakaar) [reply index.js]: @thealphadollar  catch block becomes active only when there is a netwrok error, and 4xx and 5xx aren't network errors. See [the 1st answer](https://stackoverflow.com/questions/38235715/fetch-reject-promise-and-catch-the-error-if-status-is-not-ok) and [this](https://support.loader.io/article/30-error-network-faq). What we can do it send a error message like `res.send('Unable to connect  to docs')` and  in frontend `then` block throw an error which will redirect to there `catch` block or directly show to the errror in `then` block only

**Comment by rakaar** [comment]
I have included the same changes in the #9. Hence closing this PR @thealphadollar.

**Pull request closed without merging**